### PR TITLE
[berkeley] feat: fix scripts to use different paths if mina is a submodule of o1js

### PR DESCRIPTION
**Pull request by MartinMinkov**
Explain your changes:
Fixes some of the scripts called by dune to use different absolute paths if we are building mina in the context of the o1js as mina being a submodule.

The issue occurs when we build o1js from the root directory with mina as a submodule. Dune will try to resolve some absolute paths from build scripts that will fail and not build o1js properly. To fix this, we conditionally check for submodule locations and change the paths if we have a mina submodule present, otherwise, we use the same build process. This lets us build mina the same way but also build o1js dependencies properly.

Explain how you tested your changes:
Built mina from the root directory to test normal building and built o1js using mina as a submodule successfully.

Addresses https://github.com/o1-labs/o1js/issues/1096

### CODE REVIEW

**Comment by MartinMinkov** [comment]
!ci-build-me

**Review by querolita — APPROVED** [review]

**Review by ghost-not-in-the-shell — APPROVED** [review]

**Comment by MartinMinkov** [comment]
!ci-build-me

**Comment by MartinMinkov** [comment]
!ci-build-me

File: scripts/mina-local-network/mina-local-network.sh
Comment (shimkiv) [inline scripts/mina-local-network/mina-local-network.sh:1]: Why permissions were changed for this Shell script?

**Review by shimkiv — APPROVED** [review]
- Looks good but left quick question.

File: scripts/mina-local-network/mina-local-network.sh
Comment (mitschabaude) [reply scripts/mina-local-network/mina-local-network.sh:1]: @MartinMinkov

Comment (MartinMinkov) [reply scripts/mina-local-network/mina-local-network.sh:1]: I think it changed in this [commit](https://github.com/MinaProtocol/mina/commit/d86653e713447c3471dae63f2414663ca39c5695#diff-26baf2e6b95eebc993ae2de66384050f91c02304154fc86bfe04938cbfa02c67), which was originally in the branch `rampup`

Comment (shimkiv) [reply scripts/mina-local-network/mina-local-network.sh:1]: In commit you mentioned it goes other way around: `100644 → 100755` but here it is `100755 → 100644`. Again, not a big deal just convenient to have scripts excitable by default.

Comment (MartinMinkov) [reply scripts/mina-local-network/mina-local-network.sh:1]: Ah, that's weird. I'm not sure where it changed then, this PR only stacks https://github.com/MinaProtocol/mina/pull/14239/commits/ee332bd51fff9ffa505c4b40816c8ca39ddb69a6 and https://github.com/MinaProtocol/mina/pull/14239/commits/ee332bd51fff9ffa505c4b40816c8ca39ddb69a6 on top of the merge :/

**Comment by MartinMinkov** [comment]
!ci-build-me

File: src/libp2p_ipc/dune [hunk 1]
```diff
@@ -1,7 +1,7 @@
 (library
  (name libp2p_ipc)
  (public_name libp2p_ipc)
- (flags -w -53)
+ (flags -w -53-55)
```
Comment (deepthiskumar) [inline src/libp2p_ipc/dune:4]: why is this needed?

**Comment by deepthiskumar** [comment]
!ci-build-me

File: src/libp2p_ipc/dune (hunk 1 above)
Comment (MartinMinkov) [reply src/libp2p_ipc/dune:4]: Fair question! Nix fails while building `libnp2p_ipcs`, ignoring warning `55` while building `src/libp2p_ipcs/libp2p_ipcs_capnp.ml` seems to fix it. [See this internal discussion](https://o1-labs.slack.com/archives/C03DZ0MCCFR/p1696352880596909) for more details.

**Comment by MartinMinkov** [comment]
Moved too https://github.com/MinaProtocol/mina/pull/14782
